dashboard/modules/job/job_manager.py

```diff
@@ -325,6 +326,37 @@ async def _polling(self, child_process: subprocess.Popen) -> int:
                 # still running, yield control, 0.1s by default
                 await asyncio.sleep(self.SUBPROCESS_POLL_PERIOD_S)
 
+    async def _poll_all(self, pid: int):
+        """Poll process and all its child processes until all are completed."""
+        parent_process = psutil.Process(pid)
+        children = parent_process.children(recursive=True)
+
+        while True:
```

Comment:
Any downside to moving the `parent_process` and `children` definition inside the `while` loop? It seems like this would catch more processes in case more child processes get created while waiting.

Reply:
It seems like after the parent process is killed, the child will no longer be found through `children(recursive=True)` because it will become a child of pid=1. So we probably have to find the children once in the beginning before sending the kill signals.